cache/lru_cache.cc

```diff
@@ -505,10 +505,16 @@ uint32_t LRUCache::GetHash(Handle* handle) const {
 
 void LRUCache::DisownData() {
 // Do not drop data if compile with ASAN to suppress leak warning.
+#if defined(__clang__)
+#if !defined(__has_feature) || !__has_feature(address_sanitizer)
+  shards_ = nullptr;
+#endif
+#else   // __clang__
 #ifndef __SANITIZE_ADDRESS__
   shards_ = nullptr;
-  num_shards_ = 0;
 #endif  // !__SANITIZE_ADDRESS__
+#endif  // __clang__
+  num_shards_ = 0;
```

Comment:
Should this appear together with `shards_ = nullptr`? If compiling with ASAN, we should not drop data. Namely we should not set `num_shards_` to 0. The behavior is current code is that: set `num_shards_` to 0 whether ASAN or not. Does not look like it's the same.